pkgs/applications/editors/vscode/extensions/ms-dotnettools-csharp/default.nix

```diff
@@ -1,65 +1,74 @@
-, mono
 }:
-
 let
-  # Get as close as possible as the `package.json` required version.
-  # This is what drives omnisharp.
-  rtDepsSrcsFromJson = lib.importJSON ./rt-deps-bin-srcs.json;
-
-  rtDepsBinSrcs = builtins.mapAttrs (k: v:
-      let
-        # E.g: "OmniSharp-x86_64-linux"
-        kSplit = builtins.split "(__)" k;
-        name = builtins.elemAt kSplit 0;
-        system = builtins.elemAt kSplit 2;
-      in
-      {
-        inherit name system;
-        installPath = v.installPath;
-        binaries = v.binaries;
-        bin-src = fetchurl {
-          urls = v.urls;
-          inherit (v) sha256;
-        };
-      }
-    )
-    rtDepsSrcsFromJson;
-
-  rtDepBinSrcByName = bSrcName:
-    rtDepsBinSrcs."${bSrcName}__${stdenv.targetPlatform.system}";
-
-  omnisharp = rtDepBinSrcByName "OmniSharp";
-  vsdbgs = [
-    (rtDepBinSrcByName "Debugger")
-  ] ++ lib.optionals (stdenv.isDarwin) [
-  # Include the aarch64-darwin debugger binaries on x86_64-darwin.  Even though OmniSharp will be
-  # running under Rosetta 2, debugging will fail to start if both sets of binaries are not present.
-    (rtDepsBinSrcs."Debugger__aarch64-darwin")
-  ];
-  razor = rtDepBinSrcByName "Razor";
+  vsixInfo = {
+    x86_64-linux = {
+      url = "https://github.com/OmniSharp/omnisharp-vscode/releases/download/v1.25.0/csharp-1.25.0-linux-x64.vsix";
+      sha256 = "1cqqjg8q6v56b19aabs9w1kxly457mpm0akbn5mis9nd1mrdmydl";
+      binaries = [
+        ".debugger/vsdbg-ui"
+        ".debugger/vsdbg"
+        ".omnisharp/1.39.0-net6.0/OmniSharp"
+        ".razor/createdump"
+        ".razor/rzls"
+      ];
+    };
+    aarch64-linux = {
+      url = "https://github.com/OmniSharp/omnisharp-vscode/releases/download/v1.25.0/csharp-1.25.0-linux-arm64.vsix";
+      sha256 = "0nsjgrb7y4w71w1gnrf50ifwbmjidi4vrw2fyfmch7lgjl8ilnhd";
+      binaries = [
+        ".debugger/vsdbg-ui"
+        ".debugger/vsdbg"
+        ".omnisharp/1.39.0-net6.0/OmniSharp"
+        # Linux aarch64 version has no Razor Language Server
+      ];
+    };
+    x86_64-darwin = {
+      url = "https://github.com/OmniSharp/omnisharp-vscode/releases/download/v1.25.0/csharp-1.25.0-darwin-x64.vsix";
+      sha256 = "01qn398vmjfi9imzlmzm0qi7y2h214wx6a8la088lfkhyj3gfjh8";
+      binaries = [
+        ".debugger/x86_64/vsdbg-ui"
+        ".debugger/x86_64/vsdbg"
+        ".omnisharp/1.39.0-net6.0/OmniSharp"
+        ".razor/createdump"
+        ".razor/rzls"
+      ];
+    };
+    aarch64-darwin = {
+      url = "https://github.com/OmniSharp/omnisharp-vscode/releases/download/v1.25.0/csharp-1.25.0-darwin-arm64.vsix";
+      sha256 = "020j451innh7jzarbv1ij57rfmqnlngdxaw6wdgp8sjkgbylr634";
+      binaries = [
+        ".debugger/arm64/vsdbg-ui"
+        ".debugger/arm64/vsdbg"
+        ".debugger/x86_64/vsdbg-ui"
+        ".debugger/x86_64/vsdbg"
+        ".omnisharp/1.39.0-net6.0/OmniSharp"
+        ".razor/createdump"
+        ".razor/rzls"
+      ];
+    };
+  }.${stdenv.targetPlatform.system};
```

Comment:
missing or throw unsupported platform

Reply:
I'm unsure on how to do this. Would I do something like `or error "Error message"`?